docs/wallets.md

```diff
@@ -0,0 +1,57 @@
+# For Wallet Developers
+
+codex32 is a new format for BIP32 master seeds. It is essentially a replacement for
+BIP39 or SLIP39 seed words, and the user workflow should be much the same, except
+that:
+
+* The character set is different (bech32 characters rather than a wordlist).
+* Seeds may be split across multiple shares, rather than encoded as a single string.
+* It is possible to do specific error detection/correction during entry.
+
+There are two levels of wallet support:
+
+* The ability to import seeds/shares; here we essentially just have recommendations about dealing with errors.
+* The ability to generate seeds/shares on the device; here our guidance is more involved.
+
+We encourage every wallet to support importing seeds and shares, since
+* the technical difficulty is low (roughly on par with that of supporting Segwit addresses, plus optional error-correction support);
+* the added functionality is isolated from the rest of the wallet (once the seed is imported you don't care where it came from); and
+* beyond correctness of the code, there is little risk (no need to source randomness or execute potentially variable-time algorithms).
+
+Supporting seed generation is a little more involved so the tradeoff between
+implementation complexity and user value is less clear, especially since the
+Codex provides users instructions on doing generation themselves.
+
+## Import Support
+
+codex32 shares may be any length between 128 and 512 bits.
+Wallets should support import of 128- or 256-bit seeds; other lengths are optional.
+
+128-bit seeds are 48 characters in length, including the `MS1` prefix.
+256-bit seeds are 74. For other bit-lengths, see the BIP.
+
+The process for entering shares is:
+
+1. The user should enter the first share. To the extent possible given screen limitations, when entering share data, data should be displayed in uppercase, with visually distinct four-character windows. The first four-character window should include the `MS1` prefix, which should be pre-filled.
+1. Once the first share is fully entered, the wallet should validate the checksum and header before accepting it.
+   * The user should not be able to enter mixed-case characters. The user must be able to enter all bech32 characters as well as `?` indicating an erasure. Wallets may allow users to enter non-bech32 characters, at their discretion. (This may be useful to guide error correction, e.g. by attempting to replace all `B`s with `8`s.)
+   * If the header is invalid, the wallet should highlight this and prevent the user from entering additional data until it is fixed. An invalid header is one that starts with a character other than `0` or `2` through `9`, or one which starts with `0` but whose share index is not `S`. For shares after the first, a header is also invalid if its threshold and identifier do not match those of the first share.
```

Comment:
I think Ben already mention this, but the header is covered by the error correcting checksum.  Therefore users need to be able to enter invalid header data all the way to the end so that it can be corrected.

Reply:
For subsequent shares, it should be pre-filled after 'ms1' with the first valid share's header and not editable. The entry cursor starts at position 9, share_index.

What Andrew wants to prevent is someone accidentally re-entering a previous share, as it'd be very frustrating on a ledger style input. The suggestion was a hard warning if share_index is not unique. If user assets they're entering a unique share, index would be replaced with an erasure / ?, otherwise they'd get a new share and enter that..

Regarding threshold 0 and index != s, both the threshold and the share index should be marked with an erasure (?) imo. And during error correction implementations should always substitute index 's' when checking threshold = '0'.

I'm of the opinion certain mistakes should be automatically substituted with ? as the user types, to increase their chances of understanding how the format works and correcting their mistake. With the exception of repeating a share index as that's a waste of user time, that should require confirmation to continue (marking it as an erasure).

Reply:
Inputting share_index = 's' in subsequent share entries should also always be marked as an erasure or changed to '5' (then tried as an erasure if '5' fails to correct) imo. (electronic implementations will shuffle share indexes for redundancy privacy so it's not reasonable to assume they're usually early alphabet characters)

Reply:
After the 1st valid share is accepted, the length of that share should be used to inform later deletion and insertion errors.

For known wrong strings (wrong length on subsequent, wrong charset/case or any of the erasures mentioned above) the submit button might say "attempt correction".

In my experience testing my error correcting import GUI in Bails, it's much faster to accept an incorrect length entry and let the computer attempt to correct and then verify corrections match the paper than find and fix the discrepancy yourself. So as much as possible we should allow them to make mistakes and suggest a correction.

Reply:
Will change the text to say that the wallet "SHOULD ask for confirmation" before continuing with an invalid header.